catlin/pkg/validator/task_validator_test.go

```diff
@@ -0,0 +1,80 @@
+package validator
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/tektoncd/plumbing/catlin/pkg/parser"
+	"gotest.tools/v3/assert"
+)
+
+const taskWithInvalidImageRef = `
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: invalid-image-refs
+  labels:
+    app.kubernetes.io/version: a,b,c
+  annotations:
+    tekton.dev/tags: a,b,c
+    tekton.dev/pipelines.minVersion: "0.12"
+    tekton.dev/displayName: My Example Task
+spec:
+  description: |-
+    A summary of the resource
+
+    A para about this valid task
+
+  params:
+  - name: img_ver
+    type: string
+
+  steps:
+  - name: whitelisted
+    image: ubuntu
+  - name: variables
+    image: ubuntu:$(params.img_ver)
+  - name: invalid-image
+    image: api
+  - name: missing-repo
+    image: api:1.0
+  - name: no-tags
+    image: quay.io/api
+  - name: valid
+    image: registry.io/api:1.0
+`
+
+func TestTaskValidator_InvalidImageRef(t *testing.T) {
```

Comment:
@vdemeester @chmouel Could you please `lgtm` if this test covers all/most cases? 
To me this seems to cover all cases but will wait for you to do the honours :)

Reply:
imho, this is too involved :sweat_smile: 
- I would rather not do any whitelist at all, and warn in all the case where a *one-word* image is used (be it `ubuntu`, `alpine` or `foobar`.
- according to the spec, nothing should make `quay.io/api` invalid. We **know** it is, because we know quay, but a program can't really make that assumption (and if it does, it feels weird). An exemple with https://r.j3ss.co/, the image references there are `r.j3ss.co/clisp:latest` and should be considered as valid.

Reply:
👍 on @vdemeester I dont think we should make it over complicated.